### PR TITLE
Upgrade python to v3.11.4 & invoke updates

### PR DESCRIPTION
**Pull request by zyassine**

### CODE REVIEW

File: src/noos_inv/docker.py [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
-        "repo": None,
-        "user": "AWS",
-        "token": None,
-        "arg": None,
-        # Non-sensitive
-        "file": "Dockerfile",
-        "context": ".",
-        "name": "webserver",
-        "tag": "test",
+CONFIG = Config(
```
Comment (pieeri) [inline src/noos_inv/docker.py:9]: ❓ 
Where is defined such a new API for collections in `invoke`'s documentation?

As per Invoke's configuration hierarchy:
https://docs.pyinvoke.org/en/stable/concepts/configuration.html#collection-configuration

File: src/noos_inv/docker.py [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
     }
-}
+)
 
 
 # Docker deployment workflow:
 
 
-@task
+@task()
```
Comment (pieeri) [inline src/noos_inv/docker.py:30]: 💡 

Does initiating the decorator (required for base-class declared decorators) solve the `mypy` linting issue?

**Review by pieeri — COMMENTED** [review]
Just questions at this stage.

File: src/noos_inv/docker.py (hunk 2 above)
Comment (zyassine) [reply src/noos_inv/docker.py:30]: Yes it does

File: src/noos_inv/docker.py (hunk 1 above)
Comment (zyassine) [reply src/noos_inv/docker.py]: It was necessary for tests context: https://docs.pyinvoke.org/en/stable/api/context.html

I made an update to only change the context init in tests

File: pyproject.toml [hunk 3]
```diff
@@ -5,7 +5,7 @@ build-backend = "poetry.masonry.api"
 [tool.poetry]
 # Description
 name = "noos-inv"
-version = "0.0.11"
+version = "0.0.12"
 description = "Shared workflows across CI/CD pipelines"
 # Credentials
 license = "MIT"
```
Comment (pieeri) [inline pyproject.toml:11]: 🐼 

A version left behing in the "classifiers" section of the `pyproject.toml`

**Review by pieeri — CHANGES_REQUESTED** [review]
One thing left behind at last look ; after that, good to go!

**Review by pieeri — APPROVED** [review]
👍  Nice addition, thks!

**Pull request merged**